pyfar/dsp/regularization.py

```diff
@@ -5,333 +5,373 @@
 
     def __repr__(self):
         return f"Regularization object with regularization "\
-            f"of type '{self._regu_type}'."
+            f"of type '{self._regularization_type}'."
 
     @classmethod
-    def from_frequency_range(cls, frequency_range):
+    def from_frequency_range(cls, signal, frequency_range,
+                             regularization_within=0, beta=1, target=None):
         r"""
-        Compute regularization from frequency range.
+        Regularization from a given frequency range.
 
-        Defines a frequency range within which the regularization factor is set
-        to ``0``. Outside the frequency range the regularization factor is
-        ``1`` and can be scaled using the `beta` parameter.
-        The regularization factors are cross-faded using a raised cosine
-        window function with a width of :math:`\sqrt{2}f` above and below the
-        given frequency range.
+        Defines a frequency range within which the regularization
+        :math:`\epsilon(f)` is set to ``regularization_within``.
+        Outside the frequency range the regularization is
+        :math:`\epsilon(f)=1` and can be controlled using the `beta` parameter.
+        The regularization factors are cross-faded using a raised cosine window
+        function with a width of :math:`\sqrt{2}f` above and below the given
+        frequency range.
 
         Parameters
         ----------
+        signal : Signal
+            Signal to be inverted.
         frequency_range : array like
             Array like containing the lower and upper frequency limit in Hz.
-        """
-        instance = cls(regu_type = "frequency range")
+        regularization_within: float, optional
+            Set regularization inside frequency range. The default is ``0``.
+        beta : float, string, optional
+            Beta parameter to control the scaling of the regularization as in
+            :eq:`regularized_inversion`. Can be a
+
+            ``numerical value``
+                Usually between ``0`` and ``1``, with ``0`` being no
+                regularization applied.
+            ``'energy'``
+                Normalize the regularization to match the signal's energy.
+
+                :math:`Normalization Factor = \frac{\frac{1}{N}\sum_{k=0}^{N-1}|S[k]|^2}{\frac{1}{N}\sum_{k=0}^{N-1}|\epsilon[k]|^2}`
+            ``'max'``
+                Normalize the regularization :math:`\epsilon(f)` to the maximum
+                magnitude of a given signal :math:`S(f)`.
+
+                :math:`Normalization Factor = \frac{\max(|S(f)|)}{\max(|\epsilon(f)|)}`
+            ``'mean'``
+                Normalize the regularization :math:`\epsilon(f)` to the mean
+                magnitude of a given signal :math:`S(f).`
+
+                :math:`Normalization Factor = \frac{\text{mean}(|S(f)|)}{\text{mean}(|\epsilon(f)|)}`
 
-        if len(frequency_range) != 2:
+            The default is ``1``.
```

Comment:
Just a note: In the current implementation at https://pyfar.readthedocs.io/en/stable/modules/pyfar.dsp.html#pyfar.dsp.regularized_spectrum_inversion, I think beta='max'. We should be aware of changing this behavior here

Reply:
Maybe we can discuss that briefly tomorrow.